ludwig/data/dataset/ray.py

```diff
@@ -172,7 +186,10 @@ def initialize_batcher(self, batch_size=128, should_shuffle=True, seed=0, ignore
     @lru_cache(1)
     def __len__(self):
         # TODO(travis): find way to avoid calling this, as it's expensive
-        return next(self.dataset_iter).count()
+        next_iteration_length = next(self.dataset_iter).count()
+        if self.num_dataset_partitions > 1 and self.window_status:
+            return next_iteration_length * self.num_dataset_partitions
```

Comment:
My concern with this solution is that this might not be accurate -- two partitions may not have the same number of examples.